doc/man3/OSSL_CMP_CTX_new.pod

```diff
@@ -0,0 +1,664 @@
+
+=item B<OSSL_CMP_OPT_IMPLICITCONFIRM>
+
+        Request server to enable implicit confirm mode, where the client
+        does not need to send confirmation upon receiving the
+        certificate. If the server does not enable implicit confirmation
+        in the return message, then confirmation is sent anyway.
+
+=item B<OSSL_CMP_OPT_DISABLECONFIRM>
+
+        Do not confirm enrolled certificates, to cope with broken servers
+        not supporting implicit confirmation correctly.
+B<WARNING:> This setting leads to unspecified behavior and it is meant
+exclusively to allow interoperability with server implementations violating
+RFC 4210.
+
+=item B<OSSL_CMP_OPT_UNPROTECTED_SEND>
+
+        Send messages without CMP-level protection.
+
+=item B<OSSL_CMP_OPT_UNPROTECTED_ERRORS>
+
+        Accept unprotected error responses which are either explicitly
+        unprotected or where protection verification failed. Applies to regular
+        error messages as well as certificate responses (IP/CP/KUP) and
+        revocation responses (RP) with rejection.
+B<WARNING:> This setting leads to unspecified behavior and it is meant
+exclusively to allow interoperability with server implementations violating
+RFC 4210.
+
+=item B<OSSL_CMP_OPT_IGNORE_KEYUSAGE>
+
+        Ignore key usage restrictions in signer certificate when
+        validating signature-based protection in received CMP messages.
+        Else, 'digitalSignature' must be allowed by CMP signer certificates.
+
+=item B<OSSL_CMP_OPT_PERMIT_TA_IN_EXTRACERTS_FOR_IR>
+
+        Allow retrieving a trust anchor from extraCerts and using that
+        to validate the certificate chain of an IP message.
+
+=back
+
+OSSL_CMP_CTX_get_option() reads the current of the given option
+(e.g., OSSL_CMP_OPT_IMPLICITCONFIRM) from the given OSSL_CMP_CTX structure.
+
+OSSL_CMP_CTX_set_log_cb() sets in B<ctx> the log callback function C<cb>
+for error, warning, info, debug, etc. messages.
+It fails if the OpenSSL trace facility is disabled.
+When C<cb> is NULL any output will be ignored.
+Alternatively, L<OSSL_CMP_log_open(3)> may be used to direct logging to stdout.
+
+OSSL_CMP_CTX_set_log_verbosity() is a macro setting the
+OSSL_CMP_OPT_LOG_VERBOSITY context option to the given level.
+
+OSSL_CMP_CTX_print_errors() outputs any entries in the OpenSSL error queue.
+It is similar to B<ERR_print_errors_cb()> but uses the CMP log callback function
+if set in the C<ctx> for uniformity with CMP logging if given. Otherwise it uses
+B<ERR_print_errors(3)> to print to stderr (unless OPENSSL_NO_STDIO is defined).
+
+OSSL_CMP_CTX_set1_serverPath() sets the HTTP path of the CMP server on the host.
+
+OSSL_CMP_CTX_set1_serverName() sets the given server Address (as IP or name)
+in the given OSSL_CMP_CTX structure.
+
+OSSL_CMP_CTX_set_serverPort() sets the port of the CMP server to connect to
+Port defaults to OSSL_CMP_DEFAULT_PORT = 8080 if not set explicitly.
```

Comment:
Is this a reasonable default? Why not 80? (or even 443)?

Reply:
We would like to avoid collisions with non-CMP services. But in the RFC's is no default port recommended.

Reply:
have you thought of registering an ALPN extension value, so you can use port 443?

Reply:
The "next protocol" is always HTTP, which is used to transfers CMP messages which are MIME media type "application/pkixcmp". So, I as far as I understand (I might be wrong), there's no need for a new ALPN extension value, and port 443 would already be possible.

From TLS or HTTP point of view, all the services a CA implementation provides could be served on the same port, distinguished by using either a different host name or path component in the URI. However, CA implementations might use non-default ports as they might utilize them already for other applications (management, CRL, OCSP, ...), lacking configuration options for mapping an individual host names or path to a different application.

As the relevant RFCs here are those for HTTP(S), I guess the standard-conforming way would be to use 443 for CMP/HTTP/TLS, and 80 for CMP/HTTP as defaults.

Reply:
I followed @mpeylo 's proposal and changed the OSSL_CMP_DEFAULT_PORT to 80.